felix/config/metadata.go

```diff
@@ -0,0 +1,495 @@
+// Copyright (c) 2024 Tigera, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package config
+
+import (
+	_ "embed"
+	"fmt"
+	"go/ast"
+	"go/doc"
+	"go/parser"
+	"go/token"
+	"reflect"
+	"regexp"
+	"slices"
+	"strings"
+
+	v3 "github.com/projectcalico/api/pkg/apis/projectcalico/v3"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+
+	lcconfig "github.com/projectcalico/calico/libcalico-go/config"
+	"github.com/projectcalico/calico/libcalico-go/lib/set"
+)
+
+var fieldsToIgnore = set.From(
+	// The rekey time is used by the IPsec tests, but it isn't exposed in FelixConfiguration.
+	"IPSecRekeyTime",
+)
+
+var clusterInfoFields = set.From(
+	// Internal settings from ClusterInformation
+	"ClusterGUID",
+	"ClusterType",
+	"CalicoVersion",
+	"Variant",
+	"CNXVersion",
+)
+
+var nodeFields = set.From(
+	// Moved to Node.
+	"IpInIpTunnelAddr",
+	"IPv4VXLANTunnelAddr",
+	"IPv6VXLANTunnelAddr",
+	"VXLANTunnelMACAddr",
+	"VXLANTunnelMACAddrV6",
+	"NodeIP",
+)
+
+var ConfigGroups = map[string]string{
+	"^(Datastore|Typha|Etcd|FelixHostname)": "00 Datastore connection",
+
+	"^Log":                       "00 Process: Logging",
+	"^Go":                        "00 Process: Go runtime",
+	"^Feature":                   "00 Process: Feature detection/overrides",
+	"^Health":                    "00 Process: Health port and timeouts",
+	"^Prometheus.*Metrics":       "00 Process: Prometheus metrics",
+	"^(Debug|StatsDumpFilePath)": "97 Debug/test-only (generally unsupported)",
+
+	"^(Iptables|Ipsets|KubeNodePortRanges|MaxIpsetSize)": "20 Dataplane: iptables",
+	"^Nftables": "21 Dataplane: nftables",
+	"^BPF":      "22 Dataplane: eBPF",
+	"^Windows":  "23 Dataplane: Windows",
+	"^(Openstack|Metadata|EndpointReporting|Reporting)": "25 Dataplane: OpenStack support",
+	"^(XDP|GenericXDP)": "25 Dataplane: XDP acceleration for iptables dataplane",
+
+	"^(IPv4|IPv6|)VXLAN": "31 Overlay: VXLAN overlay",
+	"^IpInIp":            "32 Overlay: IP-in-IP",
+	"^Wireguard":         "33 Overlay: Wireguard",
+	"^IPSec":             "34 Overlay: IPSec",
+
+	"^FlowLogs":       "40 Flow logs: file reports",
```

Comment:
not in OSS ;-)

Reply:
Yeah, I thought I'd just leave these here so that the ordering/naming is visible.  Not like the Enterprise docs are private?